src/test/java/org/springframework/hateoas/mvc/TypeReferencesIntegrationTest.java

```diff
@@ -82,12 +83,17 @@ public void setUp() {
 	@Test
 	public void usesResourceTypeReference() {
 
-		server.expect(requestTo("/resource")).andRespond(withSuccess(RESOURCE, MediaTypes.HAL_JSON));
+		server.expect(times(1), requestTo("/resource")).andRespond(withSuccess(RESOURCE, MediaTypes.HAL_JSON));
+		server.expect(times(1), requestTo("/resource")).andRespond(withSuccess(RESOURCE, MediaTypes.HAL_JSON_UTF8));
```

Comment:
Duplicated server expect with `MediaTypes.HAL_JSON_UTF8`. Not sure the preferred way to test the `MediaTypes.HAL_JSON` in addition to `MediaTypes.HAL_JSON_UTF8`